coreapi/main_api.py

```diff
@@ -33,6 +36,68 @@
 logger = RekogntionLogger(name="main_api")
 
 
+def text_reco(image):
+    """     Scene Text Recognition
+    Args:
+            *   image: numpy array of cropped text
+    Workflow:
+            *   A numpy array of a cropped text is taken as input
+                inference input dimension requires dimension of (100,32)
+                therefore the input is first resizing to required
+                input dimension and then normalized.
+            *   Now the processed output is further processed to make it a
+                json format which is compatible to TensorFlow Serving input.
+            *   Then a http post request is made at localhost:8501.
+                The post request contain data and headers.
+            *   Incase of any exception, it return relevant error message.
+            *   output from TensorFlow Serving is further processed using
+                wordninja and then returned as a dictionary which keeps Text
+                as key and processed output as value.
+    Returns:
+            *   Dictionary having text as Key and processed output as value.
+    """
+
+    logger.info(msg="text_reco called")
+    image = cv2.resize(image, tuple((100, 32)), interpolation=cv2.INTER_LINEAR)
+    image = np.array(image, np.float32) / 127.5 - 1.0
+    data = json.dumps({"inputs": [image.tolist()]})
+    try:
+        headers = {"content-type": "application/json"}
+        url = urllib.parse.urljoin(base_url, text_reco_url)
+        json_response = requests.post(url, data=data, headers=headers)
+    except requests.exceptions.HTTPError as errh:
+        logger.error(msg=errh)
+        return {"Error": "An HTTP error occurred."}
+    except requests.exceptions.ConnectionError as errc:
+        logger.error(msg=errc)
+        return {"Error": "A Connection error occurred."}
+    except requests.exceptions.Timeout as errt:
+        logger.error(msg=errt)
+        return {"Error": "The request timed out."}
+    except requests.exceptions.TooManyRedirects as errm:
+        logger.error(msg=errm)
+        return {"Error": "Bad URL"}
+    except requests.exceptions.RequestException as err:
+        logger.error(msg=err)
+        return {"Error": "Facial Expression Recognition Not Working"}
+    except Exception as e:
+        logger.error(msg=e)
+        return {"Error": "Facial Expression Recognition Not Working"}
+    predictions = json.loads(json_response.text).get("outputs", "error")
```

Comment:
does outputs key contain string datatype? wanted to make sure that the output variables are consistent

Reply:
Yes. All outputs take the form of a dictionary.
For no error/exception: Required key and value are string type
and for exceptions: Key is "Error" and value is a string describing the error.
Sure as pointed in the conversation below by you instead of having error here, a more specific value should be there.